### DISCUSSION TITLE
How to get Google provider ID token

### DISCUSSION THREAD

**Discussion by sjbzhc** [nextauthjs/next-auth#2428] [Help] [answered]
### Question 💬

Hi,

I am trying to log in and get an id_token.

My config is as follows:

```
const providers = [
        Providers.Google({
            clientId: process.env.GOOGLE_ID,
            clientSecret: process.env.GOOGLE_SECRET,
            idToken: true,
        }),
    ]

const pages = { signIn: "/login" };

const callbacks =  {
  async jwt(token, user, account, profile, isNewUser, idToken) {
    console.log('token :>> ', token);
    console.log('idToken :>> ', idToken);
    if (account?.accessToken) {
      token.accessToken = account.accessToken;
    }
    return token;
  }
};

const options = {
  providers,
  callbacks,
  pages
};

const auth = (req, res) => NextAuth(req, res, options);
```

I already reviewed #393 and if I understand correctly, I should be able to access the idToken in the jwt callback as described in #837.

In the [documentation](https://next-auth.js.org/configuration/providers) I also read that I can override the options of the provider, so I added idToken: true.

If I don't override tokenId to true, I am able to log in, but I have no access to the idToken.

After overriding idTioken to true, I get following error:

```
[next-auth][error][oauth_callback_handler_error]
https://next-auth.js.org/errors#oauth_callback_handler_error Error: Missing or invalid provider account
    at callbackHandler (/Users/user/projects/basic-next/node_modules/next-auth/dist/server/lib/callback-handler.js:18:119)
    at Object.callback (/Users/user/projects/basic-next/node_modules/next-auth/dist/server/routes/callback.js:100:48)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
```

### How to reproduce ☕️

Override tokenId to true in the Google provide options:

```
    Providers.Google({
      clientId: process.env.GOOGLE_ID,
      clientSecret: process.env.GOOGLE_SECRET,
      idToken: true
    }),
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by mmmoussa** [comment, marked as answer]
You can grab `id_token` from `account` in the jwt callback through `account.id_token`

**Reply by agusterodin** [reply]
I get this error if I have idToken set to true

![image](https://user-images.githubusercontent.com/10248395/138568217-7537cade-7ff9-4218-bf37-1dd362040e1c.png)

Have you experienced this?

**Comment by maquintal** [comment]
I have the same issue, but I'm using Azure Provider, and it seems like the account props into jwt is undefined, but in the signIn i can see my id_token

![image](https://user-images.githubusercontent.com/45339322/151620855-8307f8de-8f6f-4885-b4e5-d3b0ad80f497.png)

**Reply by maquintal** [reply]
okay I just resolve my issue see the solution below, FYI I have put props, only for debugging purpose

![image](https://user-images.githubusercontent.com/45339322/151621954-99014304-263d-4d36-8fee-accbc849c64d.png)
